core/dbt/config/runtime.py

```diff
@@ -77,6 +77,12 @@ def from_parts(cls, project, profile, args):
             .DEFAULTS['quote_policy']
         )
         quoting.update(project.quoting)
+        if project.archive and not allow_archive_blocks:
+            # if the user has an `archive` section, raise an error
+            raise DbtProjectError(
+                'Invalid project configuration: "archive" is not allowed'
```

Comment:
Can we link to docs here? I imagine a lot of folks who skip the release notes will hit this case. Let's do something like:
```
The `archive` section in `dbt_project.yml` is no longer supported. Please use a
`snapshot` block instead. For more information on snapshot blocks and a script
to help migrate these archives, please consult the 0.14.0 migration guide:

https://docs.getdbt.com/v0.14/docs/upgrading-to-014
```